### ISSUE TITLE
change auth policy on infra views to be chefservers instead of infra:servers

### ISSUE THREAD

**Issue by vsingh-msys** [chef/automate#3019]
In order to make infra views accessible to admin only required to move infra views only policies resource at the top level also current policies `infra:servers:*` have conflicts with ingest service policies or custom policies defined by the user.

- change auto polcies `infra:servers:*` to `chefservers:*`
- Add projects to orgs.
- Add nested action `chefserver:org:cookbook:*`

**Comment by vsingh-msys** [comment]
by changing policy with `chefservers` at the top it raises an error 

```
  rpc GetServers (infra_proxy.request.GetServers) returns (infra_proxy.response.GetServers) {
    option (google.api.http).get = "/infra/servers";
    option (chef.automate.api.policy).resource = "chefservers";
    option (chef.automate.api.policy).action = "read";
    option (chef.automate.api.iam.policy).resource = "chefservers";
    option (chef.automate.api.iam.policy).action = "chefservers:list";
  };


```

```
GEN: api/external/infra_proxy/infra_proxy.proto
[policy] couldn't apply v2 template: invalid IsAuthorizedReq.Action: 'chefservers:list' does not match regex pattern "^[a-z][a-zA-Z]*(?::[a-z][a-zA-Z]*){2}$"
```

seems that we required the next level in resource name and action

In order to do that we can think of following options



### OPTION 1 change `infra:servers` to `chef:servers` instead of `chefservers` directly

```
  rpc GetServers (infra_proxy.request.GetServers) returns (infra_proxy.response.GetServers) {
    option (google.api.http).get = "/infra/servers";
    option (chef.automate.api.policy).resource = "chef:servers";
    option (chef.automate.api.policy).action = "read";
    option (chef.automate.api.iam.policy).resource = "chef:servers";
    option (chef.automate.api.iam.policy).action = "chef:servers:list";
  };

```







### OPTION 2  change `infra:servers` to `chefservers:servers`

```
  rpc GetServers (infra_proxy.request.GetServers) returns (infra_proxy.response.GetServers) {
    option (google.api.http).get = "/infra/servers";
    option (chef.automate.api.policy).resource = "chefservers:servers";
    option (chef.automate.api.policy).action = "read";
    option (chef.automate.api.iam.policy).resource = "chefservers:servers";
    option (chef.automate.api.iam.policy).action = "chefservers:servers:list";
  };

```
@susanev & @vjeffrey nice to have ur opinion on this which sounds better?

**Comment by vjeffrey** [comment]
i would vote for option 2, but curious what other folks' opinions are

**Comment by susanev** [comment]
wait, im confused about whats going on with this error. let me chat with auth folks tomorrow and then ill get back to you.

**Comment by msorens** [comment]
The error is that a v2 action must have 3 terms: a topic, a sub-topic, and a verb (e.g. `topic:subTopic:verb`). The original sample shown above (`chefservers:list`) had only two terms, which is considered an error.

**Comment by vsingh-msys** [comment]
@msorens Thanks for the clarification that is exactly what is happening.

**Comment by susanev** [comment]
right, yah. sorry I didn't phrase my question well. I still want to chat with the auth team tomorrow to determine the best direction forward.

**Comment by vsingh-msys** [comment]
@susanev sure that would help us thank you so much.

Also planing to cover this work in already created PR https://github.com/chef/automate/pull/2908 will edit the info accordingly.

**Comment by vsingh-msys** [comment]
Closed #3019 via #2908
